source/python/neuropod/utils/eval_utils.py

```diff
@@ -29,9 +29,9 @@
 
 def check_output_matches_expected(out, expected_out):
     for key, value in expected_out.items():
-        if value.dtype.type == np.str_:
+        if value.dtype.type == np.string_ or value.dtype.type == np.str_:
             # All strings are equal
-            success_condition = (value == out[key]).all()
+            success_condition = (value.astype(np.string_) == out[key]).all()
```

Comment:
For Python 3, if value is np.str_ that is unicode string and may have symbols that can't be converted with astype(nnp.string_).

Just checked:
python3
```
>>> b = np.array("привет", dtype=np.str_)
>>> b
array('привет', dtype='<U6')
>>> b.astype(np.string_)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-5: ordinal not in range(128)
```

So, covertion from np.string_ to np.str_ would be allowed always. 

Do you expect that conversion is always possible?